### PR TITLE
Release 5.9

### PR DESCRIPTION
**Pull request by matthid**
* ENHANCEMENT: Write NUnit arguments to an arguments file, fixes problems with long command lines - https://github.com/fsharp/FAKE/pull/2114
* ENHANCEMENT: Added `SpecFlowNext` module to `Fake.DotNet.Testing.SpecFlow` with improved API and missing arguments - https://github.com/fsharp/FAKE/pull/2143
* ENHANCEMENT (BREAKING): Updated and finalized the new (and undocumented) process API which is more unit-testable - https://github.com/fsharp/FAKE/pull/2131 
* ENHANCEMENT: Updated `Fake.Testing.ReportGenerator` to include `ClassFilter` and `FileFilter` - https://github.com/fsharp/FAKE/pull/2120
* ENHANCEMENT: Improve TeamCity integrations - https://github.com/fsharp/FAKE/pull/2138
* ENHANCEMENT: Update `Fake.Tools.Pickles` to include latest CLI additions - https://github.com/fsharp/FAKE/pull/2133
* BUGFIX: Fix TeamCity (named) Artifact upload - https://github.com/fsharp/FAKE/pull/2147
* BUGFIX: AppVeyor module should not trace `appveyor.exe` calls - https://github.com/fsharp/FAKE/pull/2137
* BUGFIX: Always print the stack-trace when in verbose mode - https://github.com/fsharp/FAKE/issues/2136

### CODE REVIEW

File: src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+              | Some v ->
+                yield sprintf "--%s" paramName
+                yield v
+              | _ -> () }
+
+    let args = 
+        [
+            yield parameters.SubCommand |> string
+            
+            if not (isNull parameters.ProjectFile) then
```
Comment (matthid) [inline src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs]: @magicmonty I noticed while writing the test that `null` leads to an ugly exception. Is this parameter required? Can we give a good default>

File: src/test/Fake.Core.UnitTests/Fake.DotNet.Testing.SpecFlow.fs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+open Fake.Core
+open Fake.DotNet
+open Fake.DotNet.Testing
+open Fake.Testing
+open Expecto
+
+[<Tests>]
+let tests =
+  testList "Fake.DotNet.Testing.SpecFlow.Tests" [
+    testCase "Test that new argument generation works" <| fun _ ->
```
Comment (matthid) [inline src/test/Fake.Core.UnitTests/Fake.DotNet.Testing.SpecFlow.fs]: @magicmonty Can you take a look at this test (if it is reasonably) and maybe add some to the other PRs as well?

Comment (matthid) [reply src/test/Fake.Core.UnitTests/Fake.DotNet.Testing.SpecFlow.fs]: Do we need to run the specflow exe with mono when not on windows?

File: src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs (hunk 1 above)
Comment (magicmonty) [reply src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs]: Which one gives an exception? The SubCommand or the ProjectFile?
Both are needed. For the SubCommand the default should be "generateall" 
The Project file is needed and thus must not be null. This is the reason I added options on the SpecFlowNext variant

File: src/test/Fake.Core.UnitTests/Fake.DotNet.Testing.SpecFlow.fs (hunk 2 above)
Comment (magicmonty) [reply src/test/Fake.Core.UnitTests/Fake.DotNet.Testing.SpecFlow.fs]: @matthid 

> Do we need to run the specflow exe with mono when not on windows?
I don't know. I suppose so.

File: src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs (hunk 1 above)
Comment (matthid) [reply src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs]: Ok if it is needed can you please remove the nullcheck I added and make sure that user gets a proper error message if the parameter is not given?

File: src/test/Fake.Core.UnitTests/Fake.DotNet.Testing.SpecFlow.fs (hunk 2 above)
Comment (magicmonty) [reply src/test/Fake.Core.UnitTests/Fake.DotNet.Testing.SpecFlow.fs]: > @magicmonty Can you take a look at this test (if it is reasonably) and maybe add some to the other PRs as well?

Sure, I have a look

File: src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs (hunk 1 above)
Comment (magicmonty) [reply src/app/Fake.DotNet.Testing.SpecFlow/SpecFlowNext.fs]: @matthid 
I removed the null check and adjusted the API a bit. The ProjectFile now needs to be provided on "run". I have opened a new PR for this, as I cannot push to this branch

**Pull request merged**